addons/website/static/src/js/website.ace.js

```diff
@@ -146,33 +152,61 @@ var ViewEditor = Widget.extend({
-            .jsonRpc('/website/customize_template_get', 'call', args)
-            .then(function (views) {
-                self.loadViews.call(self, views);
-                self.open.call(self);
+            .jsonRpc('/website/get_editor_resources', 'call', args)
+            .then(function (res) {
                 var curentHash = window.location.hash;
-                var indexOfView = curentHash.indexOf("?view=");
-                if (indexOfView >= 0) {
-                    var viewId = parseInt(curentHash.substring(indexOfView + 6, curentHash.length), 10);
-                    self.$('#ace-view-list').val(viewId).change();
+                var indexOfView = curentHash.match(/.*\?view=(.*)/);
+                var viewId = false;
+                var views = res[self.editor_type]
+                if (indexOfView && indexOfView[1]) {
+                    viewId = indexOfView[1];
+                    var type = _.last(viewId.split('.'));
+                    if(type == 'css' || type == 'less'){
+                        self.editor_type = type;
+                        self.toggle_list();
+                        views = res[self.editor_type];
+                    }else{
+                        viewId = parseInt(viewId, 10);
+                    }
+                    self.$('#ace-view-list-html').val(viewId).change();
                 } else {
                     if (views.length >= 2) {
-                        var mainTemplate = views[1];
-                        self.$('#ace-view-list').val(mainTemplate.id).trigger('change');
+                        viewId = views[1].id;
                     }
-                    window.location.hash = hash;
                 }
+                self.res = res;
+                self.loadViews.call(self, views);
+                self.open.call(self);
+                self.$(self.active_list).val(viewId).trigger('change');
+                self.updateHash.call(self);
+                //Hide options from editor if server returns empty list
+                _.each(res, function(list, type){
+                    if(list.length == 0){
+                        self.$el.find('.type_switcher[data-edit-type="'+type+'"]').hide();
+                    }
+                });
             });
     },
     loadViews: function (views) {
-        var $viewList = this.$('#ace-view-list').empty();
-        _(this.buildViewGraph(views)).each(function (view) {
-            if (!view.id) { return; }
+        var self = this;
+        if(this.editor_type == 'html'){
+            var $viewList = this.$('#ace-view-list-html').empty();
+            _(this.buildViewGraph(views)).each(function (view) {
+                if (!view.id) { return; }
+                this.views[view.id] = view;
 
-            this.views[view.id] = view;
-            new ViewOption(this, view).appendTo($viewList);
-            this.loadView(view.id);
-        }.bind(this));
+                new ViewOption(this, view).appendTo($viewList);
```

Comment:
Remove ViewOption object and website.ace_view_option template. You can use:

$viewList.append(_.str.sprintf('<option value="%s">%s</option>', view.id, _.str.sprintf("%s%s", _.str.repeat("- ", view.level), view.name)));

It's shorter and clearer.